### ISSUE TITLE
Empty lines keeps growing each time the wildcard files are saved

### ISSUE THREAD

**Issue by crackfoo** [adieyal/sd-dynamic-prompts#270]
This has been happening now for quite a long time and I'm not sure if it's just me? I figured it would get reported but seems no.

Every time I save a wildcard file, the empty lines between prompts increases. If I don't keep removing them, sometimes I have to scroll pages and pages and pages to get to the next prompt.

**To Reproduce**
If I simply view the contents of a wildcard file and save it. Next time I open it, it has more empty lines between prompts.

Also mention which checkboxes have been set.

This is a local install, updated as of today but like I mentioned, I've been seeing this for quite a while...

Only things set are enabled and Don't apply to negative prompts.

**Expected behaviour**

To keep the formatting the same each time the file is saved.

**Comment by adieyal** [comment]
Does this happen when you edit the file from the Wildcard Manager?

**Comment by crackfoo** [comment]
Yes, that's what I mean, using the editor.

**Comment by adieyal** [comment]
Could you please send me an example of a file before and after you save? It doesn't happen on my end, but this might be related to Windows newlines as compared as Linux (I'm assuming your OS is Windows?)

**Comment by crackfoo** [comment]
Sure. Yes, using windows. I can attach the files if you really want them... but I simply just went into windows explorer, rclick, New->Text Document and made an empty txt file. Here I just made a new file:

<img width="466" alt="image" src="https://user-images.githubusercontent.com/6303714/220947326-2af831c5-bcb7-4d9b-a979-46927e835404.png">

And already, when I open the file this is where the lines ended up:

<img width="372" alt="image" src="https://user-images.githubusercontent.com/6303714/220947697-3eb6adae-57a5-4353-82fc-248d2041abf4.png">

And when I reload wildcards, and go back and just look at the file:

<img width="252" alt="image" src="https://user-images.githubusercontent.com/6303714/220948063-6860efcd-2291-4ff3-a665-ac237983a12f.png">

And if I don't make any change and save it again:

<img width="352" alt="image" src="https://user-images.githubusercontent.com/6303714/220948610-f1d3699d-fb30-4db7-b276-92ebad41db6f.png">

So it just continues to go and go like this...

**Comment by crackfoo** [comment]
Also... I do actually RDP to this machine... not sure if that might be important... if something wacky is happening from that?

**Comment by adieyal** [comment]
Doesn't sound like that would be the problem. Unfortunately, I'm not able to reproduce this. Perhaps someone else with a windows machine can confirm what's going on?

**Comment by crackfoo** [comment]
Yeah seems it's not. I tried adding --listen to it and load the webui from a mac machine using firefox and I still see this anomaly...

**Comment by ventolin42** [comment]
Happens to me too. Local install on Windows and i am using the --share option to access via Firefox.

**Comment by adieyal** [comment]
@ventolin42 only when using Firefox?

**Comment by crackfoo** [comment]
I see it on both firefox and chrome...

**Comment by Velour-Fog** [comment]
I also noticed this problem on windows 11. I just edit them in notepad++ instead..

**Comment by Mistborn-First-Era** [comment]
Same here on windows with google chrome. I just use vs code to edit now

**Comment by Vonpiper** [comment]
I too am having this issue.
Windows 10, Firefox

Here are before and after pictures taken with notepad++.  It looks like each time I press "save wildcards" there is an extra carriage return being inserted.

before:
![image](https://user-images.githubusercontent.com/97364305/227412367-4985aeb8-b048-4d77-9188-6377cc004130.png)

after:
![image](https://user-images.githubusercontent.com/97364305/227412516-623c3a68-1dda-4dcc-b206-759e8521b173.png)

**Comment by Vonpiper** [comment]
I tried debugging the python function save_file_callback() but I'm unable to change the behavior, even if I replace the whole function's contents with 'pass'.  I'm not sure why altering the python file doesn't change the behavior in the browser.

Anyhow, problem might be related to using os.linesep
https://stackoverflow.com/questions/4025760/python-file-write-creating-extra-carriage-return

Below screenshot is what I wanted to try:

![image](https://user-images.githubusercontent.com/97364305/227427911-0b061d88-5ed9-4ff5-a8e7-d96762185ed0.png)

**Comment by Vonpiper** [comment]
The above change works for me.  I no longer get extra new lines on windows 10.

I would create a pull request for you but I'm not sure how.

**Comment by adieyal** [comment]
Thanks - I just push an update - could you please and let me know if this issue is fixed?

**Comment by crackfoo** [comment]
I updated and now my wildcard list is blank... I did also update auto1111
from today's updates.... so perhaps they broke something?

**Comment by crackfoo** [comment]
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/8894

I get this error it seems, to perhaps the update is the culprit

**Comment by Jonseed** [comment]
I also have this issue.

**Comment by akx** [comment]
Please see https://github.com/adieyal/sd-dynamic-prompts/issues/338 – the newest a1111 updates will have broken a ton of extensions, this included.
